src/olympia/activity/tests/test_views.py

```diff
@@ -143,15 +143,22 @@ def test_deleted_version_regular_reviewer(self):
         self.version.delete()
 
         # There was a listed version, it has been deleted but still, it was
-        # there, so listed reviewers should still be able to access.
+        # there, so listed reviewers should still be able to access if they
+        # have Addons:ViewDeleted
         self._login_reviewer()
         response = self.client.get(self.url)
+        assert response.status_code == 404
+
+        user = UserProfile.objects.get(username='reviewer')
+        self.grant_permission(user, 'Addons:ViewDeleted')
```

Comment:
Removing the check on version being the latest version forced me to fix the `test_reply_to_deleted_version_is_400` test below, which was expecting a 400... but it really should have expected a 404 instead. Digging further, I found that developers and reviewers without `Addons:ViewDeleted` were able to view activities on deleted versions, which was not consistent with every other API/view. So I fixed that, which in turn forced me to adjust a few more tests.